eclair-core/src/main/scala/fr/acinq/eclair/UInt64.scala

```diff
@@ -16,40 +16,42 @@
 
 package fr.acinq.eclair
 
-import java.math.BigInteger
-
+import com.google.common.primitives.UnsignedLongs
 import scodec.bits.ByteVector
+import scodec.bits._
 
-case class UInt64(private val underlying: BigInt) extends Ordered[UInt64] {
+case class UInt64(private val underlying: Long) extends Ordered[UInt64] {
 
-  require(underlying >= 0, s"uint64 must be positive (actual=$underlying)")
-  require(underlying <= UInt64.MaxValueBigInt, s"uint64 must be < 2^64 -1 (actual=$underlying)")
+  override def compare(o: UInt64): Int = UnsignedLongs.compare(underlying, o.underlying)
 
-  override def compare(o: UInt64): Int = underlying.compare(o.underlying)
+  def toByteVector: ByteVector = this match {
+    case x if x <= UInt64(255) => ByteVector.fromLong(underlying, size = 1)
```

Comment:
A more compact way of doing this:
```scala
def toByteVector: ByteVector = underlying match {
  case 0 => hex"00"
  case _ => ByteVector.fromLong(underlying).dropWhile(_ == 0)
}
```

Reply:
Nice one! I was a bit wary of doing so many comparisons and this solves it elegantly.